src/com/google/javascript/jscomp/DefaultPassConfig.java

```diff
@@ -1726,6 +1726,7 @@ private static CompilerPass createPeepholeOptimizationsPass(
     }
     optimizations.add(new PeepholeFoldConstants(late, useTypesForOptimization));
     optimizations.add(new PeepholeCollectPropertyAssignments());
+    optimizations.add(new PeepholeCombineArrayPush());
```

Comment:
Since this optimization requires type information to work, don't add it to the list unless `useTypesForOptimization` is `true`.